user_data.sh

```diff
@@ -41,7 +41,11 @@ head -n $line $KEYS_FILE > $TEMP_KEYS_FILE
 
 # Synchronize the keys from the bucket.
 aws s3 sync --delete $BUCKET_URI $PUB_KEYS_DIR
-cat $PUB_KEYS_DIR/* >> $TEMP_KEYS_FILE
+for filename in $PUB_KEYS_DIR/*; do
+    cat $filename >> $TEMP_KEYS_FILE
+    # Ensure each new key starts on a newline
+    sed -i -e '$a\' $TEMP_KEYS_FILE
```

Comment:
This exceeds my sed fu, but seems to work. However, it would perhaps be more concise (and efficient) to do it in one go, e.g. by replacing the loop contents with `sed 's/\n?$/\n/' < $filename >> $TEMP_KEYS_FILE`?